config/src/lib.rs

```diff
@@ -1,11 +1,296 @@
-pub fn hello_world() {
-    println!("Hello World!!");
+use std::ffi::OsString;
+use std::fmt::Error;
+
+use clap::{App, Arg, ArgMatches};
+
+// Include all "stolen" ripgrep code in this module
+mod ripgrep_config;
+
+/// This trait defines the behavior that all configuration classes need to
+/// provide in order for the clap matcher to be generated from the config object
+pub trait ConfigDescription {
+    fn get_config(&self) -> Configuration;
+}
+
+/// This struct describes some properties that can be set for an application as well
+/// as the list of options that the program can understand
+/// These values (apart from the list of options) are only used to generate the
+/// console help message
+pub struct Configuration {
+    /// the name of the application
```

Comment:
Nitpick and I'm fine the way it is but it looks weird with lower case initial letters in the generated docs.

Reply:
Changed

Reply:
Unless I'm mistaken....you haven't changed it but that's fine